src/components/ec-phone-number-input/ec-phone-number-input.vue

```diff
@@ -149,197 +149,186 @@
-        return require(`svg-country-flags/png100px/${countryCode.toLowerCase()}.png`);
-      } catch (err) {
-        return null;
-      }
+  isWarning: {
+    type: Boolean,
+    default: false,
+  },
+  isMasked: {
+    type: Boolean,
+    default: false,
+  },
+  warningTooltipMessage: {
+    type: String,
+  },
+  errorMessage: {
+    type: String,
+  },
+  errorTooltipMessage: {
+    type: String,
+  },
+  countries: {
+    type: Array,
+  },
+  areCountriesLoading: {
+    type: Boolean,
+    default: false,
+  },
+  isDisabled: {
+    type: Boolean,
+    default: false,
+  },
+  disabledTooltipMessage: {
+    type: String,
+  },
+  isSensitive: {
+    type: Boolean,
+    default: false,
+  },
+  countryPlaceholder: {
+    type: String,
+  },
+  phoneNumberPlaceholder: {
+    type: String,
+  },
+  isSearchEnabled: {
+    type: Boolean,
+    default: true,
+  },
+  searchCountryPlaceholder: {
+    type: String,
+    default: 'Search...',
+  },
+  noCountriesText: {
+    type: String,
+    default: 'No results found',
+  },
+  level: {
+    type: String,
+    validator(value) {
+      return ['notification', 'modal', 'tooltip', 'level-1', 'level-2', 'level-3'].includes(value);
     },
   },
-};
+});
+
+const emit = defineEmits([
+  'update:modelValue',
+  'change',
+  'focus',
+  'open',
+  'after-open',
+  'country-change',
+  'phone-number-change',
+]);
+
+const uid = getUid();
+const countriesHasFocus = ref(false);
+const popoverOptions = ref({
+  autoSize: 'min',
+});
+const popoverWidthReference = ref(null);
+
+const id = computed(() => `ec-phone-number-input-field-${uid}`);
+const errorId = computed(() => (isInvalid.value ? `ec-phone-number-input-${uid}` : null));
+const countriesItems = computed(() => props.countries.map(country => ({
+  areaCode: country.areaCode,
+  iconPath: getCountryFlagPath(country.countryCode),
+  id: country.countryCode,
+  name: country.text,
+  text: `${country.text} ${country.areaCode}`, // the text is only displayed in the tooltip, it's just to make items searchable by area code and name of the country
+  value: country,
+})));
+const countriesModel = computed({
+  get() {
+    return props.modelValue.country;
+  },
+  set(item) {
+    emit('update:modelValue', {
+      ...props.modelValue,
+      country: item.value,
+    });
+  },
+});
+const phoneNumberModel = computed({
+  get() {
+    if (props.isMasked) {
+      return mask(props.modelValue.phoneNumber);
+    }
+
+    return props.modelValue.phoneNumber;
+  },
+  set(phoneNumber) {
+    emit('update:modelValue', {
+      ...props.modelValue,
+      phoneNumber,
+    });
+  },
+});
+const selectedCountryAreaCode = computed(() => props.modelValue?.country?.areaCode || null);
+const selectedCountryImage = computed(() => getCountryFlagPath(props.modelValue?.country?.countryCode));
+const selectedCountryName = computed(() => props.modelValue?.country?.name);
+const isInvalid = computed(() => !!props.errorMessage);
```

Comment:
This computed prop doesn't seem to be used anywhere else but one place, we can inline the logic there:
```
const errorId = computed(() => (props.errorMessage ? `ec-phone-number-input-${uid}` : null));
```

Reply:
i can see it is used on lines 38,105, 115

Reply:
I meant the `isInvalid` is not used anywhere else but in `errorId` computed prop. So we can inline it into the `errorId` computed prop.

Reply:
ah ok but also isInvalid is present on lines 116, 122, 252

Reply:
As I mentioned in Corina's PR there is no need to introduce isInvalid here as it is just renaming of errorMessage prop.

Reply:
removed isInvalid variable